src/test/java/org/rulez/demokracia/pdengine/ChoiceModifyValidationTest.java

```
@@ -0,0 +1,90 @@
+		
+		voteId = adminInfo.voteId;
+		choiceId = voteManager.addChoice(adminInfo.adminKey, adminInfo.voteId, "choice1", "user");
+		adminKey = adminInfo.adminKey;
+		choice = "New choice name";
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_voteId_is_rejected() throws ReportedException {
+		String invalidVoteId = "invalidVoteId";
+		
+		assertThrows(
+				() -> voteManager.modifyChoice(invalidVoteId, choiceId, adminKey, choice)
+			).assertMessageIs(String.format("illegal voteId: %s",invalidVoteId));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_choiceId_is_rejected() throws ReportedException {
+		String invalidChoiceId = "invalidChoiceId";
+		
+		assertThrows(
+				() -> voteManager.modifyChoice(voteId, invalidChoiceId, adminKey, choice)
+			).assertMessageIs(String.format("Illegal choiceId: %s",invalidChoiceId));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_adminKey_is_rejected() throws ReportedException {
+		String invalidAdminKey = "invalidAdminKey";
+		
+		assertThrows(
+				() -> voteManager.modifyChoice(voteId, choiceId, invalidAdminKey, choice)
+			).assertMessageIs(String.format("Illegal adminKey: %s",invalidAdminKey));
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void ok() throws ReportedException {
+		voteManager.modifyChoice(voteId, choiceId, adminKey, choice);
+		
+		ChoiceEntity choiceEntity = voteManager.getChoice(voteId, choiceId);
+		assertEquals(choiceEntity.name, choice);
+	}
+	
+	
+	@tested_feature("Manage votes")
+	@tested_operation("modify vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void ok_with_ballot() throws ReportedException {
```

the test case name is not verbose enough, please elaborate
the test case belongs to another behaviour: https://adadocs.demokracia.rulez.org/PDEngine/edemo/master/index.html#c49a56ca-86b8-459c-9848-a42d73666dcd
And the behaviour should have thrown a ReportedException instead of returning OK.

---

So I would like to test the modifyChoice method with not empty votes ballots.
I updated the test name.